Realm/RLMResults.h

```diff
@@ -20,6 +20,25 @@
 
 RLM_HEADER_AUDIT_BEGIN(nullability, sendability)
 
+/// A block type used for APIs which asynchronously return an `RLMResults`.
+typedef void(^RLMResultsCompletionBlock)(RLMResults * _Nullable, NSError * _Nullable);
+
+/**
+ Determines wait for download behavior when subscribing on RLMResults.
+ @see ``[RLMResults subscribeWithName:waitForSyncMode:onQueue:completion:]``
+*/
+typedef NS_ENUM(NSUInteger, RLMWaitForSyncMode) {
+    /// `subscribeWithName` will return once matching objects are downloaded
+    /// from the server only when the subscription is created the first time. If the
+    /// subscriptions already exists, the method will return without waiting for new downloads.
+    RLMWaitForSyncModeOnCreation,
+    /// `subscribeWithName` will always for downloads before returning.
+    /// The method will not return in this mode unless an internet connection is established or a timeout is set.
```

Comment:
This probably needs to be warned to the user on the `subscribe` APIs docstings. 
Can we have always a timeout and set a default value, the above warning to never happen

Reply:
If we make a mention of .always on that docstring, then I think we should do it for all of them. But putting all of them there would pollute if it's already explained on the enum.

The behavior is similar to asyncOpen and since asyncOpen doesn't have a default timeout (or any) I think keeping close to that behavior is nice. Or at least that's my preferance.